### DISCUSSION TITLE
Invalid arg type when reading from S3

### DISCUSSION THREAD

**Discussion by tedyu** [aws/aws-sdk-js-v3#6399] [Q&A]
We face the following exception using sdk v2 to read data from s3:
```
message:The "input" argument must be an instance of ArrayBuffer or ArrayBufferView. Received type number (123)
stack:TypeError [ERR_INVALID_ARG_TYPE]: The "input" argument must be an instance of ArrayBuffer or ArrayBufferView. Received type number (123)
at new NodeError (node:internal/errors:387:5)
at TextDecoder.decode (node:internal/encoding:417:15)
```
Here is related code:
```
    EventHelper.getS3()
      .getObject({
        Key: key,
        Bucket: bucket || EventHelper.getDefaultEventsBucket(),
      }, async (err, res) => {
        if (err) { return callback(err); }
        let streamErr;
        const raw = await consumers.text(res.Body).catch((err2) => {
```
consumers.text is where the exception comes.

If the writer to s3 uses sdk v3, I assume the interaction between v3 writer and v2 reader should work.
v3 ver: 3.50.0
v2 ver: 2.1341.0

Has anyone faced similar exception ?

**Comment by aBurmeseDev** [comment]
Hi @tedyu - I noticed you also opened a similar issue in the aws-sdk-js repository and the issue you're facing is related to the way you're trying to consume the response stream from the S3 object. To resolve this, you need to consume the response stream correctly. Here's code example we have in our [docs](https://github.com/aws/aws-sdk-js-v3?tab=readme-ov-file#streams).

```js
import { S3 } from "@aws-sdk/client-s3";

const client = new S3({});

const getObjectResult = await client.getObject({
  Bucket: "...",
  Key: "...",
});

// env-specific stream with added mixin methods.
const bodyStream = getObjectResult.Body;

// one-time transform.
const bodyAsString = await bodyStream.transformToString();

// throws an error on 2nd call, stream cannot be rewound.
const __error__ = await bodyStream.transformToString();
```
Hope it helps,
John
